var/spack/repos/builtin/packages/skit/package.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Skit(MakefilePackage):
+    """SPARSKIT:
+    A basic tool-kit for sparse matrix computations (Version 2).
+    Made by Yousef Saad, University of Minnesota.
+    """
+
+    homepage = "https://www-users.cs.umn.edu/~saad/software/SPARSKIT/"
+
+    version('develop', sha256='ecdd0a9968d6b45153a328710a42fe87600f0bba0e3c53896090b8ae1c113b7a',
+            url='http://www-users.cs.umn.edu/~saad/software/SPARSKIT/SPARSKIT2.tar.gz')
+
+    # The library uses blas routine which needs to be known when the lib is used.
+    # A dependent package should add self.spec['blas'].libs.ld_flags
+    # at the end of its link line.
+    # But, asis, this packages compiles without needing to know about it.
+    # depends_on('blas', type='run')
+
+    variant('pic', default=True,
+            description='Compile with position independent code.')
+    variant('debug', default=False,
+            description='Builds a debug version of the library')
+
+    # We provide the standard Make flags here:
+    # https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=flag_handler#compiler-flags
+    def flag_handler(self, name, flags):
+        spec = self.spec
+        if '+pic' in spec:
+            if name == 'fflags':
+                flags.append(self.compiler.fc_pic_flag)
+        if name == 'fflags':
+            if 'gfortran' in self.compiler.fc:
+                flags.append('-std=legacy')
+                flags.append('-Wall')
+        if '+debug' in spec:
+            if '-g' in self.compiler.debug_flags:
+                flags.append('-g')
+            if '-O0' in self.compiler.opt_flags:
+                flags.append('-O0')
+            elif '-O' in self.compiler.opt_flags:
+                flags.append('-O')
+        else:
+            if '-O3' in self.compiler.opt_flags:
+                flags.append('-O3')
+            elif '-O2' in self.compiler.opt_flags:
+                flags.append('-O2')
+
+        return (None, flags, None)
+
+    def edit(self, spec, prefix):
+        mkfile = FileFilter('makefile')
+        mkfile.filter(r'^(F90|F77|FC).*=.+', r'\1={0}'.format(self.compiler.fc))
```

Comment:
This shouldn't be necessary. Spack defines compiler wrappers `fc`, `cc`, etc. that inject the flags it needs (such as RPATH, compiler flags, default optimizations, other dependencies). By setting to `self.compiler.fc` you're replacing the spack compiler (which automatically calls the "correct" compiler) with the original compiler. This isn't what you want.

Reply:
OK I understand. I modified accordingly, but now instead of the hard set 'F77 = gfortran' in the makefile I set
F77 ?= self.compiler.fc : this way first I can change the compiler using spack (without doing anything it always compiled with gfortran even while %intel !) and second the makefile is usable outside of spack which I like when something goes wrong. I prefer that over commenting like '# F77 = gfortran'.

Reply:
I did not manage to understand what makes this makefile stubborn as the word is used in
https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html#edit-makefile
but it is !
I added some comments in the package file - and also corrected "sparsekit" to "sparskit".

Reply:
Aha, yes as you point out in the example: it should be literally `fc`, not `gfortran` nor the value `self.compiler.fc`. The `fc` script is the spack compiler wrapper for fortran.

Reply:
OK, so I removed that filter and compile with make('F77=fc').

Reply:
Ugh, so it turns out that `fc` is a `bash` builtin that doesn't play well with makefiles 😢 I'll submit an issue for that (or at least a PR update to the docs noting the issue with `fc` in makefiles). Can you change `fc` to `f77`? Hopefully that will fix. Sorry, this is a learning experience for me too.

Reply:
Oh actually, it looks like there's an even better way: change to `spack_fc`.

See https://spack.readthedocs.io/en/latest/packaging_guide.html#filtering-functions

Reply:
https://github.com/spack/spack/pull/23856

Reply:
Change is done ; we are converging !